### ISSUE TITLE
botocore is 36Mb and causes our lambda deployment to be huge

### ISSUE THREAD

**Issue by andy-d** [boto/botocore#1629]
Our code uses boto3 to do stuff, and is deployed as a lambda. In looking into our package size, I've noticed that botocore is almost all of it. 36mb out of 40mb total.

 (1) Is something stupid about our workflow here? Do I have access to AWS low level operations without using boto3?
 (2) Is there anything I can cut out of botocore that is a large amount of it?

Thanks!

**Comment by kadrach** [comment]
Out of curiosity, why do you package botocore? The Python runtime of AWS Lambda contains boto3 and botocore.

**Comment by andy-d** [comment]
@kadrach yes I realized that after filing this, sorry for the noise I'll kill it now.

**Comment by stealthycoin** [comment]
@andy-d you could kill any subdirectory in `botocore/data` if you are not using that client.

**Comment by andy-d** [comment]
@stealthycoin thanks, I already closed this as botocore and friends are all included by default so I didn't need to package them anyway.

**Comment by stealthycoin** [comment]
@andy-d I figured it could still be useful to know if you ever need to bundle a newer version than lambda has provided, as they tend to lag ~1 month behind at the moment.

**Comment by andy-d** [comment]
Oh yeah perfect thanks, that makes sense.

**Comment by dnp1** [comment]
Still does not understand why so many examples files are package alongside with this simple package, my docker images are larger due to it, my lambdas are larger (I need aiobotocore and the versions are not properly synced with aws lambda version, so I need to include it)

There's a reason to package data directory?

**Comment by pauldraper** [comment]
_Laughs in 67MB_

**Comment by MT-Cash** [comment]
https://stackoverflow.com/questions/75367378/is-it-possible-to-shrink-botocore-within-site-packages might be interesting for people comming here

**Comment by gdamore** [comment]
botocore is still friggin' ginormous, and I don't know why.  But this is blowing up cloud images -- e.g. this is by *far* the largest package in my Alpine Linux distribution.

**Comment by kartik-nighania** [comment]
@gdamore  you already have it available in the execution environment. https://docs.aws.amazon.com/lambda/latest/dg/lambda-python.html

**Comment by gdamore** [comment]
In my case this isn't a lambda deployment.  Rather these are containers deployed via LXC and Docker. Our customers run both on-prem and in the cloud (and hybrid cloud) and botocore is the largest single component in the docker image. Over 60 MB(!) for a python library that speaks to couple of AWS services?

**Comment by d3netxer** [comment]
it is good to know that, unless you need a specific version, you do not need to include boto3 in your deployments because it is already provided with the runtime environment : )

**Comment by samspycher** [comment]
This seems relevant:

https://www.serverlessops.io/blog/aws-lambda-and-python-boto3-bundling
https://docs.aws.amazon.com/lambda/latest/dg/python-handler.html#python-handler-best-practices:~:text=Control%20the%20dependencies%20in%20your%20function%27s%20deployment%20package.

tl;dr – Shipping your own dependencies is a best practice recommendation from AWS themselves.
